### PR TITLE
Addition of Multi-Group mode to theory manual

### PR DESCRIPTION
**Pull request by nelsonag**
I was on a plane for a very long time today.  As a result, we get this PR :-).  
This PR incorporates #595, the inclusion of multi-group mode in to the theory manual.

### CODE REVIEW

**Comment by paulromano** [comment]
:+1: for documentation! I was also on a plane for a very very long time but all I did was watch movies. Way to make me feel bad @nelsonag.

File: docs/source/methods/introduction.rst [hunk 1]
```diff
@@ -105,13 +114,20 @@ proceed. The life of a single particle will proceed as follows:
...
-     of type :math:`(n,xn)` are treated as scattering and the weight of the
-     particle is increased by the multiplicity of the reaction. The particle
-     then continues from step 3. If the reaction is absorption or fission, the
-     particle dies and if necessary, fission sites are created and stored in the
-     fission bank.
+     energy and angle is sampled from the appropriate distribution.  In
+     continuous-energy simulation, reactions of type :math:`(n,xn)` are treated
+     as scattering and any additional particles which may be created are added
+     to a secondary particle bank to be tracked later. In a multi-group
+     simulation, this secondary bank is ont used but the particle weight is
```
Comment (paulromano) [inline docs/source/methods/introduction.rst]: is not used

File: docs/source/methods/physics.rst [hunk 2]
```diff
@@ -4,6 +4,12 @@
 Physics
 =======
 
+There are limited differences between physics treatments used in the
+continuous-energy and multi-group modes.  If distinctions are necessary, each
+of the following sections will provide an explanation of the differences.
+Otherwise, replacing any references of the particle's energy (`E`) with
+references to the particle's energy group (`g) will suffice.
```
Comment (paulromano) [inline docs/source/methods/physics.rst]: Forgot a backtick after g

**Comment by nelsonag** [comment]
Whoops, thanks for catching those typos @paulromano! Pushing now. 

And dont feel bad about watching movies on the plane... I didnt plan ahead and was to cheap to buy the movies from the plane's wifi :-)

**Comment by paulromano** [comment]
Thanks @nelsonag!

**Pull request merged**